go/libraries/doltcore/sqle/dfunctions/dolt_checkout.go

```diff
@@ -139,7 +139,9 @@ func getRevisionForRevisionDatabase(ctx *sql.Context, dbName string) (string, st
 	return provider.GetRevisionForRevisionDatabase(ctx, dbName)
 }
 
-func checkoutRemoteBranch(ctx *sql.Context, dbName string, dbData env.DbData, roots doltdb.Roots, branchName string) error {
+// checkoutRemoteBranch checks out a remote branch creating a new local branch with the same name as the remote branch
+// and set its upstream. The upstream persists out of sql session.
+func checkoutRemoteBranch(ctx *sql.Context, dbName string, dbData env.DbData, branchName string) error {
```

Comment:
Beautiful work getting rid of that `roots` param and simplifying this function a little bit! 👏  The beautiful godocs are a cherry on top. 😄